SPIRE Statefulset usage prevents clusterspiffeid webhook

Steps to reproduce

1. In the helm charts use `server: kind: statefulset`
2. Run `make k8s-manifests-update VERSION=0.24.5` to create k8s manifests
3. (assuming using Minikube) `make k8s-delete`
4. `make k8s-start`
5. `eval $(minikube -p minikube docker-env`
6. `make build-local` (go make a sandwich)
7. `make deploy-local`

Expectation:

Deployment should finish without an issue.

What happens:

We get the following error:

```text
Error from server (InternalError): error when creating "./k8s/0.24.5/local/vsecm-distroless.yaml": Internal error occurred: failed calling webhook "vclusterspiffeid.kb.io": failed to call webhook: Post "https://spire-controller-manager-webhook-service.spire-system.svc:443/validate-spire-spiffe-io-v1alpha1-clusterspiffeid?timeout=10s": dial tcp 10.109.59.20:443: connect: connection refused
Error from server (InternalError): error when creating "./k8s/0.24.5/local/vsecm-distroless.yaml": Internal error occurred: failed calling webhook "vclusterspiffeid.kb.io": failed to call webhook: Post "https://spire-controller-manager-webhook-service.spire-system.svc:443/validate-spire-spiffe-io-v1alpha1-clusterspiffeid?timeout=10s": dial tcp 10.109.59.20:443: connect: connection refused
Error from server (InternalError): error when creating "./k8s/0.24.5/local/vsecm-distroless.yaml": Internal error occurred: failed calling webhook "vclusterspiffeid.kb.io": failed to call webhook: Post "https://spire-controller-manager-webhook-service.spire-system.svc:443/validate-spire-spiffe-io-v1alpha1-clusterspiffeid?timeout=10s": dial tcp 10.109.59.20:443: connect: connection refused
make: *** [makefiles/VSecMDeploy.mk:54: deploy-local] Error 1
```

Other Details:

The issue does not present itself whein using `kind: deployment`

--- a/helm-charts/0.24.5/charts/spire/templates/spire-server.yaml
+++ b/helm-charts/0.24.5/charts/spire/templates/spire-server.yaml
@@ -177,6 +177,9 @@ metadata:
     app: spire-server
     app.kubernetes.io/component: server
 spec:
+  {{- if eq .Values.server.kind "statefulset" }}
+  serviceName: spire-server
+  {{- end }}
   replicas: {{ .Values.replicaCount }}
   selector:
     matchLabels:

